### PR TITLE
Add new RocksDB options for controlling index and filter block caching

### PR DESCRIPTION
**Pull request by dothebart**

### CODE REVIEW

File: 3.7/release-notes-new-features37.md [hunk 1]
```diff
@@ -525,6 +525,19 @@ new user key. Similarly the new option `--rocksdb.encryption-keyfolder` can be u
 to supply multiple user keys. A random internal key will be generated and
 encrypted with each of the provided user keys.
 
+### rocksdb storage engine options exposed
```
Comment (jsteemann) [inline 3.7/release-notes-new-features37.md:528]:
```suggestion
### RocksDB storage engine options exposed
```

File: 3.7/release-notes-new-features37.md [hunk 2]
```diff
@@ -525,6 +525,19 @@ new user key. Similarly the new option `--rocksdb.encryption-keyfolder` can be u
...
+* Multiple more rocksdb configuration options will now be exposed to be configurable
```
Comment (jsteemann) [inline 3.7/release-notes-new-features37.md:529]:
```suggestion
* Multiple additional RocksDB configuration options are now exposed to be configurable
```

File: 3.7/release-notes-new-features37.md [hunk 3]
```diff
@@ -525,6 +525,19 @@ new user key. Similarly the new option `--rocksdb.encryption-keyfolder` can be u
...
+  via arangodb:
```
Comment (jsteemann) [inline 3.7/release-notes-new-features37.md:530]:
```suggestion
in _arangod:
```

File: 3.7/release-notes-new-features37.md [hunk 4]
```diff
@@ -525,6 +525,19 @@ new user key. Similarly the new option `--rocksdb.encryption-keyfolder` can be u
...
+  * `--rocksdb.cache-index-and-filter-blocks` to make the RocksDB block cache quota
+    also include RocksDB memtable sizes
+  * `--rocksdb.cache-index-and-filter-blocks-with-high-priority` to use cache index
+    and filter blocks with high priority making index and filter blocks be less
+    likely to be evicted than data blocks
+  * `--rocksdb.pin-l0-filter-and-index-blocks-in-cache` make filter and index blocks
+    be pinned and only evicted from cache when the table reader is freed
+  * `--rocksdb.pin-top-level-index-and-filter` make the top-level index of partitioned
+    filter and index blocks pinned and only be evicted from cache when the table reader is freed
+
```
Comment (jsteemann) [inline 3.7/release-notes-new-features37.md:540]:
```suggestion
Multiple additional RocksDB configuration options are now exposed to be configurable
in _arangod_:

* `--rocksdb.cache-index-and-filter-blocks` to make the RocksDB block cache quota
  also include RocksDB memtable sizes
* `--rocksdb.cache-index-and-filter-blocks-with-high-priority` to use cache index
  and filter blocks with high priority making index and filter blocks be less
  likely to be evicted than data blocks
* `--rocksdb.pin-l0-filter-and-index-blocks-in-cache` make filter and index blocks
  be pinned and only evicted from cache when the table reader is freed
* `--rocksdb.pin-top-level-index-and-filter` make the top-level index of partitioned
  filter and index blocks pinned and only be evicted from cache when the table reader is freed

```

**Comment by Simran-B** [comment]
These options should be added to the reduce memory tutorial:
https://www.arangodb.com/docs/stable/tutorials-reduce-memory-footprint.html

**Pull request merged**